src/scikit_build_core/builder/get_requires.py

```diff
@@ -66,6 +67,25 @@ def _load_scikit_build_settings(
     return SettingsReader.from_file("pyproject.toml", config_settings).settings
 
 
+class RootPathResolver:
+    """Handle ``{root:uri}`` like formatting similar to ``hatchling``."""
+
+    def __init__(self, path: Path | None = None) -> None:
+        self.path = path if path else Path()
```

Comment:
```suggestion
@dataclasses.dataclass(frozen=True)
class RootPathResolver:
    """Handle ``{root:uri}`` like formatting similar to ``hatchling``."""

    path: Path = dataclasses.field(default_factory=Path)
```

This isn't much shorter, but it does have a nicer repr.